cmd/signature-v4-utils.go

```diff
@@ -146,16 +146,22 @@ func sumHMAC(key []byte, data []byte) []byte {
 // extractSignedHeaders extract signed headers from Authorization header
 func extractSignedHeaders(signedHeaders []string, r *http.Request) (http.Header, APIErrorCode) {
 	reqHeaders := r.Header
+	reqQueries := r.URL.Query()
 	// find whether "host" is part of list of signed headers.
 	// if not return ErrUnsignedHeaders. "host" is mandatory.
 	if !contains(signedHeaders, "host") {
 		return nil, ErrUnsignedHeaders
 	}
 	extractedSignedHeaders := make(http.Header)
 	for _, header := range signedHeaders {
+		canonicalHeaderKey := http.CanonicalHeaderKey(header)
 		// `host` will not be found in the headers, can be found in r.Host.
 		// but its alway necessary that the list of signed headers containing host in it.
-		val, ok := reqHeaders[http.CanonicalHeaderKey(header)]
+		val, ok := reqHeaders[canonicalHeaderKey]
+		if !ok {
+			// try to set headers from Query String
+			val, ok = reqQueries[strings.ToLower(canonicalHeaderKey)]
```

Comment:
What is the reason to look for `canonicalHeaderKey` in lower-case in the `reqQueries` map? Shouldn't we look for the signed header with the same case?

Reply:
I think we needn't to use CanonicalHeaderKey for this case at all. We just need to check if we can get the header value from query string. Fixed, thanks.